synthesizer/src/vm/helpers/rewards.rs

```diff
@@ -51,27 +51,46 @@ pub fn staking_rewards<N: Network>(
         return stakers.clone();
     }
 
+    // First, validate all validators and create a map of valid validators.
+    let valid_validators: IndexMap<_, _> = committee
+        .members()
+        .iter()
+        .filter_map(|(validator, (validator_stake, _is_open, commission_rate))| {
+            // Skip validators with invalid commission rates.
+            if *commission_rate > 100 {
+                error!("Commission rate ({commission_rate}) is greater than 100 - skipping validator {validator}");
+                return None;
+            }
+            // Skip validators with more than 25% of total stake.
+            if *validator_stake > committee.total_stake().saturating_div(4) {
+                trace!("Validator {validator} has more than 25% of the total stake - skipping all stakers");
+                return None;
+            }
+            Some((*validator, (*validator_stake, *commission_rate)))
+        })
+        .collect();
+
+    // Track validators not in committee.
+    // Pre-allocating with an expected capacity prevents reallocation while the mutex is held.
+    let hashset_capacity = committee.members().len();
+    let missing_validators =
+        std::sync::Mutex::new(std::collections::HashSet::<Address<N>>::with_capacity(hashset_capacity));
```

Comment:
At 100k stakers, a BTreeSet will be meaningfully faster than a HashSet

Reply:
is hashing this expensive here? otherwise a hashset should be faster for both insertions and lookups.